src/content/tutorials/build-a-slackbot/index.md

```diff
@@ -0,0 +1,696 @@
+
+<Aside>
+Note that you’re able to use great JS features like async/await inside of your Workers application, thanks to Workers’ V8 runtime: since <code>r.route</code> returns a Promise, you can write <code>await r.route(request)</code> to set <code>response</code> to the result of the resolved Promise.
+</Aside>
+
+If there is no matching route (for instance, if someone requests the path `/admin`), the function should return a response with a status code of 404. `handleRequest` checks to see if `response` is `undefined`, and if it is, it sets `response` to a new `Response` with the body text “Not found”, and a status code of 404.
+
+Finally, the function returns the `response`, whether it’s a match from the router, or a 404, back to the `fetch` event. The result will be _either_ a handled API route, or a plain HTTP response.
+
+This request/response pattern makes it really straightforward to understand _how_ requests are routed in your Workers application. You’re _almost_ done with this file: to complete it, you need to actually define the corresponding function handlers for your routes. In this tutorial, you’ll define those handlers in `src/handlers`:
+
+```sh
+---
+header: Create new folders and files
+---
+$ mkdir -p src/handlers
+$ touch src/handlers/lookup.js
+$ touch src/handlers/webhook.js
+```
+
+With those files created (you’ll fill them in soon), let’s import them at the top of `index.js`. The final version of the code looks like this:
+
+```javascript
+---
+filename: index.js
+highlight: [1, 2, 11, 12]
+---
+import lookup from './src/handlers/lookup'
+import webhook from './src/handlers/webhook'
+import Router from './router'
+
+addEventListener('fetch', event => {
+  event.respondWith(handleRequest(event.request))
+})
+
+async function handleRequest(request) {
+  const r = new Router()
+  r.post('/lookup', lookup)
+  r.post('/webhook', webhook)
+
+  let response = await r.route(request)
+
+  if (!response) {
+    response = new Response('Not found', { status: 404 })
+  }
+
+  return response
+}
+```
+
+### Creating the “lookup” route
+
+In `src/handlers/lookup.js`, you’ll define your first route handler. The `lookup` handler is a function with one argument, the `request` being passed from the `fetch` event in `index.js`:
+
+```javascript
+---
+filename: src/handlers/lookup.js
+---
+export default async request => {}
+```
+
+To understand how you should design this function, you need to understand how Slack _slash commands_ send data to URLs.
+
+According to the [documentation for Slack slash commands](https://api.slack.com/slash-commands), Slack sends an HTTP POST request to your specified URL, with a `application/x-www-form-urlencoded` content type. For instance, if someone were to type `/issue cloudflare/wrangler#1`, you could expect a data payload in the format:
+
+```
```

Comment:
Use `txt` here if you want plain formatting (rather than leaving it up to chance).